translators/src/Grammar.hs

```diff
@@ -47,8 +48,9 @@ data LocalDefn
 data Tm
   = PCon Name [Tm]        -- (parameterized) type constructor
   | DCon Name [Tm]        -- dependent type constructor (note that a dependent type is also parameterized)
-  | Arr Tm Tm             -- function type
-  | Index [Name] Tm
+  | KCon KnownT [Tm]      -- built-in (like Vec)
+  | Arr (Arg [Name] Tm) Tm           -- (non-dependent) function type
```

Comment:
Would be good to leave a note here as to why we treat dependent and non-dependent functions differently.